python/cudf/cudf/utils/utils.py

```diff
@@ -3,12 +3,10 @@
 import functools
```

Comment:
Can't comment at the top of this file, but should we take this opportunity to make this `cudf._utils` so as to not give the impression that this is part of the public API?

Reply:
Yes, I can do that. Also `cudautils`->`_cudautils`?

Reply:
I think we can move the entire `cudf.utils` module to `cudf._utils`?

Reply:
Yep that makes sense. If you don't mind, though, I'll do that in a follow-up PR because that will probably require changing the imports in a large number of files and will be easier to review in a single quick PR rather than as part of the more substantial changes here.